pyface/font.py

```diff
@@ -0,0 +1,494 @@
+            'italic bold 14pt Helvetica, Arial, sans-serif' or
+            '36pt "Comic Sans"'
+        """
+        return cls(**parse_font_description(description))
+
+    @classmethod
+    def from_toolkit(cls, toolkit_font):
+        """ Create a Font from a toolkit font object.
+
+        Parameters
+        ----------
+        toolkit_font : any
+            A toolkit font to be converted to a corresponding class instance,
+            within the limitations of the options supported by the class.
+        """
+        from pyface.toolkit import toolkit_object
+        toolkit_font_to_properties = toolkit_object(
+            'font:toolkit_font_to_properties')
+
+        return cls(**toolkit_font_to_properties(toolkit_font))
+
+    def to_toolkit(self):
+        """ Create a toolkit font object from the Font instance.
+
+        Returns
+        -------
+        toolkit_font : any
+            A toolkit font which matches the property of the font as
+            closely as possible given the constraints of the toolkit.
+        """
+        from pyface.toolkit import toolkit_object
+        font_to_toolkit_font = toolkit_object('font:font_to_toolkit_font')
+
+        return font_to_toolkit_font(self)
+
+    def __str__(self):
+        terms = []
+        if self.style != 'normal':
+            terms.append(self.style)
+        terms.extend(
+            variant for variant in variants
+            if variant in self.variants
+        )
+        if self.weight != 'normal':
+            terms.append(self.weight)
+        if self.stretch != 'normal':
+            terms.append(self.stretch)
+        size = self.size
+        # if size is an integer
+        if int(size) == size:
+            size = int(size)
+        terms.append("{}pt".format(size))
+        terms.append(
+            ', '.join(
+                repr(family) if ' ' in family else family
+                for family in self.family
+            )
+        )
+        return ' '.join(terms)
+
+    def __repr__(self):
+        traits = self.trait_get(self.editable_traits())
+        trait_args = ', '.join(
+            "{}={!r}".format(name, value)
+            for name, value in traits.items()
+        )
+        return "{}({})".format(self.__class__.__name__, trait_args)
+
+    def __eq__(self, other):
+        if isinstance(other, Font):
+            return (
+                self.family == other.family
+                and self.weight == other.weight
+                and self.stretch == other.stretch
+                and self.style == other.style
+                and self.size == other.size
+                and self.variants == other.variants
+            )
+        else:
+            return NotImplemented
+
+
+class PyfaceFont(TraitType):
+    """ A trait that holds a Pyface Font.
+
+    The value can be assigned as a string, in which case it is parsed
+    as a font description and an appropriate font created for it.
+    """
```

Comment:
I disagree strongly on this.  The `FontDescription` can only be initialized by a string but holds a `Font`, so if you had:
```
class MyClass(HasTraits):
    label_font = FontDescription("Arial")
```
you can't do:
```
foo = MyClass(label_font="Comic Sans")
```
unless you also modify the validator to also do parsing, but then you get the weird cases of
```
foo.label_font = foo.label_font
```
raising a `TraitError` unless you _also_ accept `Font` objects as valid, and then why shouldn't you be able to say
```
class MyClass(HasTraits):
    label_font = FontDescription(Font(family=["Arial"]))
```
when you can say
```
foo.label_font = Font(family=["Arial"])
```
And doing a sub-class
```
class MySubClass(MyClass):
    label_font = ????  # what is valid here?
```
is even weirder.

Basically `PyfaceFont` is what you get when you want to support text-to-Font conversion (and we do!).  There is still `Instance(Font)` if you _don't_ want string text conversion.

Reply:
My bad for omitting the `validate` in my `FontDescription`. My apologies. I was mainly thinking about how we could make `__init__` simpler and reuse what is already available from `Instance`.  So I omitted the `validate` due to laziness. For a trait type responsible for text-to-font conversion, I would still expect the text-to-font conversion to be included in the `validate` as well.

I see that if one wants to support text-to-font conversion in defining a trait, the trait type needs to support what `Instance(Font)` also supports, and the validation will have to be the combination of the two.

I have to admit I am not fond of the text-to-font conversion support. I find defining `Font(family=["Arial"], variants={"small-caps"}, weight="bold")` safe, readable, intuitive and explicit. I can read the trait class definition and then I know what to write, I don't have to learn a relaxed CSS-like language. In other words, the convenience of typing less does not give me enough confidence and I would still prefer defining `Font` traits explicitly. Therefore the conversion layer strikes me as a rather costly thing to support.

I am probably alone in that, so I will continue to review accepting this will be a popular feature.